frb_rust/src/ffi/mod.rs

```diff
@@ -55,3 +57,28 @@ pub fn wire2api_uuids(ids: Vec<u8>) -> Vec<uuid::Uuid> {
         .map(wire2api_uuid_ref)
         .collect::<Vec<uuid::Uuid>>()
 }
+
+/// Equivalent to a [`Arc::clone()`], but directly in terms of raw pointers.
+///
+/// # Safety
+///
+/// This function should never be called manually.
+extern "C" fn share_arc<T>(ptr: *const c_void) -> *const c_void {
+    unsafe {
+        Arc::<T>::increment_strong_count(ptr as _);
+        ptr
+    }
+}
+
+/// Dropper opaque data.
+///
+/// # Safety
+///
+/// This function should never be called manually.
+extern "C" fn drop_arc<T>(ptr: *const c_void) {
```

Comment:
https://api.dart.dev/stable/2.18.4/dart-ffi/NativeFunction-class.html
as I understand it for ffi.NativeFunction (native implementation) requires `extern C`, in case of web implementation it is not required